packages/svelte/src/reactivity/set.js

```diff
@@ -73,65 +73,71 @@ export class ReactiveSet extends Set {
 
 	/** @param {T} value */
 	has(value) {
-		var s = this.#sources.get(value);
+		var s = this.#tracked.get(value);
 
 		if (s === undefined) {
-			// We should always track the version in case
-			// the Set ever gets this value in the future.
-			get(this.#version);
-
-			return false;
+			s = source(super.has(value));
+			this.#tracked.set(value, s);
 		}
 
+		effect(() => () => {
```

Comment:
Not sure we want to invoke an effect as part of this method. It will add considerable overhead.

Reply:
Sure. I'm not that familiar with internals. Is there a better way to register a cleanup for a running effect?

Reply:
I'm not sure this approach of needing to clean up is the right one. Why not keep the versioning logic from before?

Reply:
The versioning logic is still there for non fine-grained updates. Signals are now created only per user's request (e.g. with `.has`). They should be cleaned up when they are no longer used anymore, though

Reply:
This is only required if .has is called from an effect. If we can somehow hook onto the parent's effect cleanup logic, that would be great